velox/functions/prestosql/tests/ArithmeticTest.cpp

```diff
@@ -821,5 +821,44 @@ TEST_F(ArithmeticTest, wilsonIntervalUpper) {
   EXPECT_DOUBLE_EQ(wilsonIntervalUpper(1, 3, kInf).value(), 1.0);
 }
 
+TEST_F(ArithmeticTest, cosineSimilarity) {
+  const auto testCosineSimilarity =
+      [&](const std::vector<
+              std::vector<std::pair<std::string, std::optional<double>>>>& left,
+          const std::vector<
+              std::vector<std::pair<std::string, std::optional<double>>>>&
+              right,
+          double expected) {
+        auto leftMap = makeMapVector<std::string, double>(left);
+        auto rightMap = makeMapVector<std::string, double>(right);
+        EXPECT_DOUBLE_EQ(
```

Comment:
Move EXPECT_DOUBLE_EQ to the caller to provide more useful error messages. Have this function return the result instead.

```
EXPECT_DOUBLE_EQ(
   (2.0 * 3.0) / (std::sqrt(5.0) * std::sqrt(10.0)),
   cosineSimilarity({{{"a", 1}, {"b", 2}}},...)
)
```